### PR TITLE
✏️ #API_Vehicles update function & news

### PR DESCRIPTION
**Pull request by K4VK4Z**
Create new functions : 

```lua
local vehicleFuel = API_Vehicles:getFuelTank(vehicleId) -- return fuel tank

local vehicleSpeed = API_Vehicles:getSpeed(vehicleId) -- return vehicle speed in km/h

API_Vehicles:setFuel(vehicleId, fuel -- Int value) 
```

### CODE REVIEW

File: src/api/vehicles/client/setFuel.lua [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+---@return number
+---@public
+
+function API_Vehicles:setFuel(vehicleId, fuel)
+    if (vehicleId) then
+        SetVehicleFuelLevel(vehicleId, fuel)
+    end
+    _NCS:trace('Unable to find vehicle', 1)
+end
+
```
Comment (MoskalykA) [inline src/api/vehicles/client/setFuel.lua:13]:
```suggestion
---setFuel
---@param vehicleId number
---@param fuel number
---@return number
---@public
function API_Vehicles:setFuel(vehicleId, fuel)
    if (vehicleId) then
        SetVehicleFuelLevel(vehicleId, fuel)
    end
    _NCS:trace('Unable to find vehicle', 1)
end
```

File: src/api/vehicles/client/getSpeed.lua [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+---@public
+
+function API_Vehicles:getSpeed(vehicleId)
+    if (vehicleId) then
+        local speed <const> = (GetEntitySpeed(vehicleId) * 3.6) --Set speed to km/h
+        return speed
+    end
+    _NCS:trace('Unable to find vehicle', 1)
+end
+
```
Comment (MoskalykA) [inline src/api/vehicles/client/getSpeed.lua:13]:
```suggestion
---getSpeed
---@param vehicleId number
---@return number
---@public
function API_Vehicles:getSpeed(vehicleId)
    if (vehicleId) then
        local speed <const> = (GetEntitySpeed(vehicleId) * 3.6) --Set speed to km/h
        return speed
    end
    _NCS:trace('Unable to find vehicle', 1)
end
```

File: src/api/vehicles/client/getFuelTank.lua [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+---@return number
+---@public
+
+function API_Vehicles:getFuelTank(vehicleId)
+    if (vehicleId) then
+        local fuel <const> = GetVehicleHandlingFloat(vehicleId, "CHandlingData", "fPetrolTankVolume")
+        return fuel
+    end
+    _NCS:trace('Unable to find vehicle', 1)
+end
```
Comment (MoskalykA) [inline src/api/vehicles/client/getFuelTank.lua:12]:
```suggestion
---getFuelTank
---@param vehicleId number
---@return number
---@public
function API_Vehicles:getFuelTank(vehicleId)
    if (vehicleId) then
        local fuel <const> = GetVehicleHandlingFloat(vehicleId, "CHandlingData", "fPetrolTankVolume")
        return fuel
    end
    _NCS:trace('Unable to find vehicle', 1)
end
```

**Review by MoskalykA — CHANGES_REQUESTED** [review]

File: src/api/objects/server/createObject.lua [hunk 4]
```diff
@@ -4,14 +4,16 @@
 ---@param y number
 ---@param z number
 ---@param heading number
+---@param freeze boolean
 ---@return void
```
Comment (MoskalykA) [inline src/api/objects/server/createObject.lua:8]: Wrong type

File: src/api/ped/server/spawnPed.lua [hunk 5]
```diff
@@ -7,16 +7,15 @@
 ---@param callback function
 ---@return void
```
Comment (MoskalykA) [inline src/api/ped/server/spawnPed.lua:8]: Wrong type

File: src/api/vehicles/client/getFuelTank.lua [hunk 6]
```diff
@@ -0,0 +1,11 @@
+---getFuelTank
+---@param vehicleId number
+---@return number
+---@public
+function API_Vehicles:getFuelTank(vehicleId)
+    if (vehicleId) then
+        local fuel <const> = GetVehicleHandlingFloat(vehicleId, "CHandlingData", "fPetrolTankVolume")
+        return fuel
+    end
+    _NCS:trace('Unable to find vehicle', 1)
```
Comment (MoskalykA) [inline src/api/vehicles/client/getFuelTank.lua:10]: Use " please

File: src/api/vehicles/client/getSpeed.lua [hunk 7]
```diff
@@ -0,0 +1,11 @@
+---getSpeed
+---@param vehicleId number
+---@return number
+---@public
+function API_Vehicles:getSpeed(vehicleId)
+    if (vehicleId) then
+        local speed <const> = (GetEntitySpeed(vehicleId) * 3.6) --Set speed to km/h
+        return speed
+    end
+    _NCS:trace('Unable to find vehicle', 1)
```
Comment (MoskalykA) [inline src/api/vehicles/client/getSpeed.lua:10]: Use " please

File: src/api/vehicles/client/setFuel.lua [hunk 8]
```diff
@@ -0,0 +1,11 @@
+---setFuel
+---@param vehicleId number
+---@param fuel number
+---@return number
+---@public
+function API_Vehicles:setFuel(vehicleId, fuel)
+    if (vehicleId) then
+        SetVehicleFuelLevel(vehicleId, fuel)
+    end
+    _NCS:trace('Unable to find vehicle', 1)
```
Comment (MoskalykA) [inline src/api/vehicles/client/setFuel.lua:10]: Use " please

**Review by MoskalykA — CHANGES_REQUESTED** [review]

**Comment by K4VK4Z** [comment]
**Hi!** 
_Modifications have been made & add/modification of divers api functions_

File: src/api/ped/client/getDistance.lua [hunk 9]
```diff
@@ -0,0 +1,9 @@
+---getDistance
+---@param pedIn number
+---@param coords table
+---@return distance
+---@public
+function API_Player:getDistance(pedId, coords)
+    local distance <const> =  #(API_Ped:getPosition(pedId) - coords)
+	return distance
+end
```
Comment (MoskalykA) [inline src/api/ped/client/getDistance.lua:9]:
```suggestion
---getDistance
---@param pedId number
---@param coords table
---@return distance
---@public
function API_Player:getDistance(pedId, coords)
    local distance <const> =  #(API_Ped:getPosition(pedId) - coords)
    return distance
end
```

**Review by MoskalykA — CHANGES_REQUESTED** [review]
There are lots of indentation errors, please correct this.

File: src/api/vehicles/client/getVehicleProperties.lua [hunk 10]
```diff
@@ -3,95 +3,146 @@
...
+        modLivery = modLivery,
+        modLightbar = GetVehicleMod(vehicleEntity, 49),
+        windows = vehicleDamage.windows,
+        doors = vehicleDamage.doors,
+        tyres = vehicleDamage.tyres,
+        leftHeadlight = vehicleDamage.leftHeadlight,
+        rightHeadlight = vehicleDamage.rightHeadlight,
+        frontBumper = vehicleDamage.frontBumper,
+        rearBumper = vehicleDamage.rearBumper,
+    }
```
Comment (MoskalykA) [inline src/api/vehicles/client/getVehicleProperties.lua:147]:
```suggestion
```

File: src/api/vehicles/client/getVehicleProperties.lua [hunk 11]
```diff
@@ -3,95 +3,146 @@
 ---@return table
 ---@public
 function API_Vehicles:getProperties(vehicleEntity)
```
Comment (MoskalykA) [inline src/api/vehicles/client/getVehicleProperties.lua:5]: vehicleEntity -> vehicleId

File: src/api/vehicles/client/setVehicleProperties.lua [hunk 12]
```diff
@@ -37,14 +38,28 @@ function API_Vehicles:setProperties(vehicleEntity, props)
         SetVehicleDirtLevel(vehicleEntity, props.dirtLevel + 0.0)
     end
     if props.color1 then
-        SetVehicleColours(vehicleEntity, props.color1, colorSecondary)
-    end
-    if props.color2 then
-        SetVehicleColours(vehicleEntity, props.color1 or colorPrimary, props.color2)
-    end
+		if type(props.color1) == 'number' then
```
Comment (MoskalykA) [inline src/api/vehicles/client/setVehicleProperties.lua:41]: ' -> "

File: src/api/vehicles/client/setVehicleProperties.lua [hunk 13]
```diff
@@ -14,6 +14,7 @@ function API_Vehicles:setProperties(vehicleEntity, props)
     local colorPrimary, colorSecondary = GetVehicleColours(vehicleEntity)
```
Comment (MoskalykA) [inline src/api/vehicles/client/setVehicleProperties.lua:14]: Review syntax and variable name

**Review by MoskalykA — CHANGES_REQUESTED** [review]

**Pull request merged**